src/main/java/net/neoforged/neoforge/common/extensions/IAttributeExtension.java

```diff
@@ -0,0 +1,217 @@
+     */
+    default MutableComponent toBaseComponent(double value, double entityBase, boolean merged, TooltipFlag flag) {
+        Attribute attr = self();
+
+        Component debugInfo = CommonComponents.EMPTY;
+
+        if (flag.isAdvanced() && !merged) {
+            // Advanced Tooltips cause us to emit the entity's base value and the base value of the item.
+            debugInfo = Component.literal(" ")
+                    .append(Component.translatable("neoforge.adv.base", FORMAT.format(entityBase), FORMAT.format(value - entityBase)).withStyle(ChatFormatting.GRAY));
+        }
+
+        MutableComponent comp = Component.translatable("attribute.modifier.equals.0", FORMAT.format(value), Component.translatable(attr.getDescriptionId()));
+
+        return comp.append(debugInfo);
+    }
+
+    /**
+     * Returns the color used by merged attribute modifiers. Only used when {@link NeoForgeMod#enableMergedAttributeTooltips()} is active.
+     * <p>
+     * Similarly to {@link Attribute#getStyle(boolean)}, this method should return a color based on the attribute's {@link Sentiment}.
+     * The returned color should be distinguishable from the color used by {@link Attribute#getStyle(boolean)}.
+     * 
+     * @param positive If the attribute modifier value is positive or not.
+     */
+    TextColor getMergedStyle(boolean isPositive);
+
+    /**
+     * Certain attributes, such as Attack Damage, are increased by an Enchantment that doesn't actually apply
+     * an attribute modifier.<br>
+     * This method allows for including certain additional variables in the computation of "base" attribute values.
+     *
+     * @param stack The stack in question.
+     * @return Any bonus value to be applied to the attribute's value, after all modifiers have been applied.
+     * 
+     * @apiNote This method is experimental as it only exists to support displaying Sharpness in item tooltips, which is currently broken by MC-271840
+     */
+    @ApiStatus.Experimental
+    default double getBonusBaseValue(ItemStack stack) {
+        // if (this == Attributes.ATTACK_DAMAGE) return EnchantmentHelper.getDamageBonus(stack, MobType.UNDEFINED);
+        return 0;
+    }
+
+    /**
+     * This method is invoked when {@link #getBonusBaseValue(ItemStack)} returns a value higher than zero.<br>
+     * It is responsible for adding tooltip lines that explain where the bonus values from {@link #getBonusBaseValue(ItemStack)} are from.
+     *
+     * @param stack   The stack in question.
+     * @param tooltip The tooltip consumer.
+     * @param flag    The tooltip flag.
+     * 
+     * @apiNote This method is experimental as it only exists to support displaying Sharpness in item tooltips, which is currently broken by MC-271840
+     */
+    @ApiStatus.Experimental
+    default void addBonusTooltips(ItemStack stack, Consumer<Component> tooltip, TooltipFlag flag) {
+        // if (this == Attributes.ATTACK_DAMAGE) {
```

Comment:
Why are these commented out lines here? If it is to show an example of what the javadocs are talking about, then could we move it to a code block in the javadocs?

Reply:
same as https://github.com/neoforged/NeoForge/pull/1551#discussion_r1770243520